racket-cas/racket-cas.rkt

```diff
@@ -741,103 +758,118 @@
 (module+ test (check-equal? (simplify '(+ 3 (* 2 (expt 8 1/2))))
                             (⊕ (⊗ 2 2 (Sqrt 2)) 3)))
 
-; combine (Maxima) : a/c + b/c = (a+b)/c
-(define (combine expr)
-  (when debugging? (displayln (list 'combine expr)))
-  (define c combine)
-  (math-match expr
-              [(⊕ (⊘ u w) (⊘ v w))         (together expr)]
-              [(⊕ u v) (let ([cv (c v)])
-                         (cond [(equal? v cv)    (⊕  u  cv)]     ; Trival case
-                               [else          (c (⊕ u cv))]))] ; May match special cases after inner combination.
-              [_ expr]))
 
-(module+ test
-  (check-equal? (combine (⊕ (⊘ (⊕ x) z) (⊘ (⊕ y x) z) (⊘ 1 z)))
-                            '(* (expt z -1) (+ 1 (* 2 x) y)))
-  (check-equal? (combine (⊕ (⊘ (⊕ x) 3) (⊘ (⊕ y x) 3) (⊘ 1 3)))
-                            '(* (expt 3 -1) (+ 1 (* 2 x) y)))
-  )
+;;;
+;;; Quotients
+;;;
+
+; For a rational number a/b the numerator and denominator are a and b respectively. 
+; For an expression u which happen be a ratio (a product of a rational and factors
+; of the form (expt u p) we compute the numerator and denominator.
+; For other expressions 1 is returned as a denominator.
+
+; When matching we will in some situations want to match a "real" quotient,
+; i.e. one where the denominator isn't 1. So we will define two different match
+; expanders ⊘ and Quotient. Here ⊘  matches only if the denominator isn't 1,
```

Comment:
In my PR, ⊘ will not match a Sum. (with term-numerator/denominator)

Looks ⊘ becomes more aggressive in your changes, we need to double check this when it is used as a match pattern.

Reply:
Hi @soegaard 
I have reviewed your changes, the main concern is about the Quotient.
No together conducted for it, is it what we need?

Reply:
I think it is a good idea to make the pattern matchers as fast as possible, so
users don't accidently use an expensive operation.
In the case of Quotient, a user can call `together` explicitly if he needs to.

But we could add, say, QuotientTogether and have both options?